setup.py

```diff
@@ -1,31 +1,98 @@
+"""
+jupyter-server-proxy setup
+"""
+import json
+from pathlib import Path
+
 import setuptools
-from glob import glob
-
-with open("README.md") as f:
-    readme = f.read()
-
-setuptools.setup(
-    name="jupyter-server-proxy",
-    version='1.6.0',
-    url="https://github.com/jupyterhub/jupyter-server-proxy",
-    author="Ryan Lovett & Yuvi Panda",
-    author_email="rylo@berkeley.edu",
-    license="BSD 3-Clause",
-    description="Jupyter server extension to supervise and proxy web services",
-    long_description=readme,
+from jupyter_packaging import (
+    combine_commands,
+    create_cmdclass,
+    ensure_targets,
+    install_npm,
+    skip_if_exists,
+)
+
+HERE = Path(__file__).parent.resolve()
+
+# The name of the project
+name = "jupyter_server_proxy"
+
+lab_path = HERE / name / "labextension"
+
+# Representative files that should exist after a successful build
+jstargets = [
+    str(lab_path / "package.json"),
+]
+
+package_data_spec = {
+    name: ["*"],
+}
+
+labext_name = "jupyter_server_proxy"
+
+data_files_spec = [
+    ("share/jupyter/labextensions/%s" % labext_name, str(lab_path), "**"),
+    ("share/jupyter/labextensions/%s" % labext_name, str(HERE), "install.json"),
+    (
+        "etc/jupyter/jupyter_server_config.d",
+        "jupyter_server_proxy/etc",
+        "jupyter-server-proxy-jupyterserverextension.json",
+    ),
+    (
+        "etc/jupyter/jupyter_notebook_config.d",
+        "jupyter_server_proxy/etc",
+        "jupyter-server-proxy-notebookserverextension.json",
+    ),
+    (
+        "etc/jupyter/nbconfig/tree.d",
+        "jupyter_server_proxy/etc",
+        "jupyter-server-proxy-nbextension.json",
+    ),
+]
+
+cmdclass = create_cmdclass(
+    "jsdeps", package_data_spec=package_data_spec, data_files_spec=data_files_spec
+)
+
+js_command = combine_commands(
+    install_npm(HERE / "jupyterlab-server-proxy", build_cmd="build:prod", npm=["jlpm"]),
+    ensure_targets(jstargets),
+)
+
+is_repo = (HERE / ".git").exists()
+if is_repo:
+    cmdclass["jsdeps"] = js_command
+else:
+    cmdclass["jsdeps"] = skip_if_exists(jstargets, js_command)
+
+long_description = (HERE / "README.md").read_text()
+
+# Get the package info from package.json
+pkg_json = json.loads((HERE / "jupyterlab-server-proxy" / "package.json").read_bytes())
+
+setup_args = dict(
+    name=name,
+    version=pkg_json["version"],
+    url=pkg_json["homepage"],
+    author=pkg_json["author"],
+    description=pkg_json["description"],
+    license=pkg_json["license"],
+    long_description=long_description,
     long_description_content_type="text/markdown",
+    cmdclass=cmdclass,
     packages=setuptools.find_packages(),
-    install_requires=['notebook', 'simpervisor>=0.4', 'aiohttp'],
-    python_requires='>=3.5',
+    install_requires=["jupyterlab~=3.0", "notebook", "simpervisor>=0.4", "aiohttp"],
```

Comment:
Done in https://github.com/jupyterhub/jupyter-server-proxy/pull/245/commits/48eec7295499ce6a0e72348a16fd77146cae9d4d and https://github.com/jupyterhub/jupyter-server-proxy/pull/245/commits/f3210df7cdb7ec7397e33c80e455e6f978951d1a. I've decided to set the requirement to `jupyterlab>=2` but consider that a placeholder until we know which versions should be supported.

I also had to install `jupyterlab` explicitly during the test actions. I hardcoded v3 there but maybe it would be nice to test in a matrix there (https://github.com/jupyterhub/jupyter-server-proxy/pull/245/commits/005c0aa729ed6083677dedd1836713a024875161 and https://github.com/jupyterhub/jupyter-server-proxy/pull/245/commits/419c1b731bfea25f6e6900fccc7b6e7505da7497)